paasta_tools/cli/cmds/status.py

```diff
@@ -385,38 +400,42 @@ def report_status_for_cluster(
             output.append('    Git sha:    None (not deployed yet)')
 
     return_code = 0
+    ssh_instances = [i for i in deployed_instances if i not in http_only_instances]
     if len(deployed_instances) > 0:
-        if use_api_endpoint:
+        if use_api_endpoint or http_only_instances:
             return_codes = [
                 paasta_status_on_api_endpoint(
-                    cluster,
-                    service,
-                    deployed_instance,
-                    system_paasta_config,
+                    cluster=cluster,
+                    service=service,
+                    instance=deployed_instance,
+                    output=output,
+                    system_paasta_config=system_paasta_config,
                     verbose=verbose,
                 )
                 for deployed_instance in deployed_instances
+                if (deployed_instance in http_only_instances or use_api_endpoint)
             ]
             if any(return_code != 200 for return_code in return_codes):
                 return_code = 1
-        else:
-            return_code, status = execute_paasta_serviceinit_on_remote_master(
-                'status', cluster, service, ','.join(deployed_instances),
+        if not use_api_endpoint and ssh_instances or not http_only_instances:
+            ssh_return_code, status = execute_paasta_serviceinit_on_remote_master(
+                'status', cluster, service, ','.join(ssh_instances),
                 system_paasta_config, stream=False, verbose=verbose,
                 ignore_ssh_output=True,
             )
+            return_code = ssh_return_code + return_code
```

Comment:
is this right? so if we get exit 1 and 1, the exit code will show as 2?

Reply:
Hmm, yeah let me try something else